src/main/java/com/r307/arbitrader/service/TradingService.java

```diff
@@ -219,12 +220,17 @@ private void enterPosition(Spread spread) {
             return;
         }
 
+        //Adjust the volume after slip so the trade stays market neutral
+        shortVolume = getVolumeForShortTrade(maxExposure, shortLimitPrice, shortScale);
+        longVolume = getVolumeForLongTrade(shortVolume, shortFeePercent, longFeePercent, longScale);
+
         // we need to add fees for exchanges where feeComputation is set to CLIENT
         final BigDecimal longVolumeWithFees = addFees(spread.getLongExchange(), spread.getCurrencyPair(), longVolume);
         final BigDecimal shortVolumeWithFees = addFees(spread.getShortExchange(), spread.getCurrencyPair(), shortVolume);
 
         // Before executing the order we adjust the step size for each side of the trade (long and short).
         // This will be the amount we sent in the execute order request to the exchange
+        // TODO the volumes need to be adjusted so the trade stays market neutral
```

Comment:
Is this still a TODO? It looks like the lines following it are making the adjustment.

Reply:
This is still a TODO. The next line only adjust the orders to the echange steps, they don't ensure that shortVolume = longVolume * feeFactor